gusto/physics.py

```diff
@@ -765,3 +767,201 @@ def evaluate(self, x_in, dt):
             self.tau.assign(dt)
         self.water_v.assign(x_in.split()[self.Vv_idx])
         self.source.assign(self.source_interpolator.interpolate())
+
+
+class SWSaturationAdjustment(Physics):
+    """
+    Represents the process of adjusting water vapour and cloud water according
+    to a saturation function, via condensation and evaporation processes.
+
+    This physics scheme follows that of Zerroukat and Allen (2020).
+
+    """
+
+    def __init__(self, equation, saturation_curve, L=None,
+                 time_varying_saturation=False, vapour_name='water_vapour',
+                 cloud_name='cloud_water', convective_feedback=False,
+                 beta1=None, thermal_feedback=False, beta2=None, gamma_v=1,
+                 time_varying_gamma_v=False, tau=None,
+                 parameters=None):
+        """
+        Args:
+            equation (:class: 'PrognosticEquationSet'): the model's equation
+            saturation_curve (ufl expression or :class: `function`): the
+                curve which dictates when phase changes occur. In a saturated
+                atmosphere vapour above the saturation curve becomes cloud,
+                and if the atmosphere is sub-saturated and there is cloud
+                present cloud will become vapour until the saturation curve is
+                reached. The saturation curve is either prescribed or
+                dependent on a prognostic field.
+            time_varying_saturation (bool, optional): set this to True if the
+                saturation curve is changing in time. Defaults to False.
+            L (float, optional): The air expansion factor multiplied by the
```

Comment:
Should we call this `L_v` instead of `L`?

Reply:
Yes, I think that makes more sense.